airflow-core/src/airflow/jobs/scheduler_job_runner.py

```diff
@@ -431,8 +432,8 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session) -
             timer.start()
 
             try:
-                query = with_row_locks(query, of=TI, session=session, skip_locked=True)
-                task_instances_to_examine: list[TI] = session.scalars(query).all()
+                locked_query = with_row_locks(query, of=TI, session=session, skip_locked=True)
+                task_instances_to_examine: list[TI] = list(session.scalars(locked_query).all())
```

Comment:
@Dev-iL Why do we need the explicit list here? Isn't that what `.all()` returns already?